cmd/version.go

```diff
@@ -17,58 +17,91 @@ limitations under the License.
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
+	"runtime"
 	"runtime/debug"
 )
 
 const unknown = "unknown"
 
-// var needs to be used instead of const as ldflags is used to fill this
-// information in the release process
+// These are filled via ldflags during build
 var (
 	kubeBuilderVersion      = unknown
 	kubernetesVendorVersion = "1.33.0"
 	goos                    = unknown
 	goarch                  = unknown
-	gitCommit               = "$Format:%H$" // sha1 from git, output of $(git rev-parse HEAD)
-
-	buildDate = "1970-01-01T00:00:00Z" // build date in ISO8601 format, output of $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+	gitCommit               = "$Format:%H$"
+	buildDate               = "1970-01-01T00:00:00Z"
 )
 
-// version contains all the information related to the CLI version
-type version struct {
+// VersionInfo holds all CLI version-related information
+type VersionInfo struct {
```

Comment:
```suggestion
// version holds all CLI version-related information
type version struct {
```

Can we avoid externalising this one?
If we externalise it (e.g., by using v instead of V), it becomes accessible outside of Kubebuilder and effectively becomes part of its public API.

We intentionally do not want to expose this structure for use by other projects. Let's keep it internal to maintain tighter control and avoid unintended usage.